packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -96,6 +97,68 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
   bool shouldRelayout(MultiChildLayoutDelegate oldDelegate) => false;
 }
 
+class _FloatingActionButtonTransition extends StatefulComponent {
+  _FloatingActionButtonTransition({
+    Key key,
+    this.child
+  }) : super(key: key) {
+    assert(child != null);
+  }
+
+  final Widget child;
+
+  _FloatingActionButtonTransitionState createState() => new _FloatingActionButtonTransitionState();
+}
+
+class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTransition> {
+  final Performance hidePerformance = new Performance(duration: _kFloatingActionButtonSegue);
+  final Performance showPerformance = new Performance(duration: _kFloatingActionButtonSegue);
+  Widget oldChild;
+
+  void initState() {
+    super.initState();
+    showPerformance.play();
+  }
+
+  void dispose() {
+    hidePerformance.stop();
+    showPerformance.stop();
+    super.dispose();
+  }
+
+  void didUpdateConfig(_FloatingActionButtonTransition oldConfig) {
+    if (config.child.key == oldConfig.child.key)
```

Comment:
We really should use a function defined in framework.dart for this purpose.  The function should check both the key and the runtimeType.